libaleth-interpreter/VMCalls.cpp

```diff
@@ -147,11 +138,24 @@ void VM::caseCreate()
         auto off = static_cast<size_t>(initOff);
         auto size = static_cast<size_t>(initSize);
 
-        msg.input_data = &m_mem[off];
-        msg.input_size = size;
+        bytes saltAndInputData;
+        if (m_OP == Instruction::CREATE)
+        {
+            msg.input_data = &m_mem[off];
+            msg.input_size = size;
+            msg.kind = EVMC_CREATE;  // FIXME: In EVM-C move the kind to the top.
+        }
+        else
+        {
+            assert(m_OP == Instruction::CREATE2);
+            // Pass salt data through EVMC as 32-byte prefix in evmc_message::input_data
+            saltAndInputData = toBigEndian(salt) + bytesConstRef(&m_mem[off], size);
```

Comment:
If we go with this, it must be documented in evmc properly.

I think it is a stretch, however uncontrollably growing `evmc_message` isn't a better option.

Reply:
Because using the message input to pass the salt requires a copy of the init code, I'm more into adding the salt param to the message.

More generic approach would be to ask the VM to come up with the address and put it in the `message.destination`.

Reply:
I'll change it to salt as a separate member of `evmc_message` then.